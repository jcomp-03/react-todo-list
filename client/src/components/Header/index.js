//  import React library
import React from "react";

function Header({ hideTodoBoard, setHideTodoBoard }) {

  const handleClick = () => {
    hideTodoBoard ? setHideTodoBoard(false) : setHideTodoBoard(true);
  };

  return (
    <header className="header">
      <span className="header__span">My ToDo List</span>
      <nav className="header__nav">
        <button className="btn header__btn" onClick={handleClick}>
          {hideTodoBoard ? "Show Tasks List" : "Hide Tasks List"}
        </button>
      </nav>
    </header>
  );
}

export default Header;
