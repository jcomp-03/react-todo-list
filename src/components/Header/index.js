//  import React library
import React from "react";

function Header() {

  return (
    <header className="header">
      <span className="header__span">My ToDo List</span>
      <nav className="header__nav">
        <button className="btn header__btn">Show Tasks Lists</button>
      </nav>
    </header>
  );
}

export default Header;
