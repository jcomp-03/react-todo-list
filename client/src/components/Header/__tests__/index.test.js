import React from "react";
import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import Header from "..";

afterEach(cleanup);

// Test #1
it("renders successfully", () => {
  render(<Header />);
});

// Test #2
it('matches snapshot DOM node structure', () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});