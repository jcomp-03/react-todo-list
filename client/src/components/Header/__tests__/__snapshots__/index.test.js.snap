// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <span
      class="header__span"
    >
      My ToDo List
    </span>
    <nav
      class="header__nav"
    >
      <button
        class="btn header__btn"
      >
        Hide Tasks List
      </button>
    </nav>
  </header>
</DocumentFragment>
`;
